File: core 2/src/main/java/com/dev/core/service/task/TaskAttachmentService.java
package com.dev.core.service.task;

import com.dev.core.model.task.TaskAttachmentDTO;
import org.springframework.web.multipart.MultipartFile;

import java.util.List;

public interface TaskAttachmentService {

    TaskAttachmentDTO uploadAttachment(Long taskId, MultipartFile file, String description, String visibility);

    List<TaskAttachmentDTO> getAttachmentsByTask(Long taskId);

    TaskAttachmentDTO getAttachmentById(Long id);

    void deleteAttachment(Long id);

    String getDownloadUrl(Long id);
}

File: CORE-FE/src/services/taskAttachment.service.ts
import axiosInstance from '../axiosInstance';
import type { TaskAttachmentDTO } from '../types/task.types';

const ATTACHMENT_API_BASE = '/tasks';

export const taskAttachmentService = {
    // --------------------------------------------------------------
    // UPLOAD ATTACHMENT
    // --------------------------------------------------------------
    uploadAttachment: async (
        taskId: number,
        file: File,
        description?: string,
        visibility: string = 'INTERNAL'
    ): Promise<TaskAttachmentDTO> => {
        const formData = new FormData();
        formData.append('file', file);
        if (description) {
            formData.append('description', description);
        }
        formData.append('visibility', visibility);

        const response = await axiosInstance.post(
            `${ATTACHMENT_API_BASE}/${taskId}/attachments`,
            formData,
            {
                headers: {
                    'Content-Type': 'multipart/form-data',
                },
            }
        );
        return response.data.data;
    },

    // --------------------------------------------------------------
    // GET ATTACHMENTS BY TASK
    // --------------------------------------------------------------
    getAttachmentsByTask: async (taskId: number): Promise<TaskAttachmentDTO[]> => {
        const response = await axiosInstance.get(`${ATTACHMENT_API_BASE}/${taskId}/attachments`);
        return response.data.data;
    },

    // --------------------------------------------------------------
    // DELETE ATTACHMENT
    // --------------------------------------------------------------
    deleteAttachment: async (taskId: number, attachmentId: number): Promise<void> => {
        await axiosInstance.delete(`${ATTACHMENT_API_BASE}/${taskId}/attachments/${attachmentId}`);
    },
};

export default taskAttachmentService;
